### ISSUE TITLE
Adding ndjson.reader & ndjson.writer

### ISSUE THREAD

**Issue by Yomguithereal** [rhgrant10/ndjson#4]
Hello @rhgrant10,

I would like to open a PR on your repo to add two things I find myself recoding often and wanted to know whether you would consider those features useful/relevant for your library: a ndjson writer & reader, having a very similar interface as the standard python ones from the `csv` module. Indeed, those would be very useful when reading the input line by line and when outputing the result in a stream fashion.

Here is an example of the API:

```python
import ndjson

# Reader
with open('./posts.jsonl') as f:
  reader = ndjson.reader(f)

  for post in reader:
    print(post['id'])

# Writer
with open('./posts.jsonl', 'w') as f:
  writer = ndjson.writer(f)

  for item in data:
    writer.writerow(item)
```

What do you think?

**Comment by rhgrant10** [comment]
Yeah I like the idea :)

**Comment by rhgrant10** [comment]
Available in [PyPI v0.3.0](https://pypi.org/project/ndjson/0.3.0).

**Comment by Yomguithereal** [comment]
Thanks @rhgrant10
